MetadataExtractor/IO/BufferReader.cs

```diff
@@ -135,4 +138,250 @@ public string GetString(int bytesRequested, Encoding encoding)
 
         return encoding.GetString(bytes);
     }
+
+    public StringValue GetNullTerminatedStringValue(int maxLengthBytes, Encoding? encoding = null, bool moveToMaxLength = false)
+    {
+        var bytes = GetNullTerminatedBytes(maxLengthBytes, moveToMaxLength);
+
+        return new StringValue(bytes, encoding);
+    }
+
+    public byte[] GetNullTerminatedBytes(int maxLengthBytes, bool moveToMaxLength = false)
+    {
+        // The number of non-null bytes
+        int length;
+
+        byte[] buffer;
+
+        if (moveToMaxLength)
+        {
+            buffer = GetBytes(maxLengthBytes);
+            length = Array.IndexOf(buffer, (byte)'\0') switch
+            {
+                -1 => maxLengthBytes,
+                int i => i
+            };
+        }
+        else
+        {
+            buffer = new byte[maxLengthBytes];
+            length = 0;
+
+            while (length < buffer.Length && (buffer[length] = GetByte()) != 0)
+                length++;
+        }
+
+        if (length == 0)
+            return [];
+        if (length == maxLengthBytes)
+            return buffer;
+        var bytes = new byte[length];
+        if (length > 0)
+            Array.Copy(buffer, bytes, length);
+        return bytes;
```

Comment:
This now lives in two places. I wonder if we can avoid that duplication somehow. Nothing jumps out at me though.

Reply:
There's a few allocations here that can be killed as well. On my list to be refactored in a another PR (along with returning a ReadOnlySpan<byte>). I'll see if we can share any of that logic with the other implementation when I do.